opentelemetry/src/test/scala/zio/telemetry/opentelemetry/TracingTest.scala

```diff
@@ -73,6 +73,95 @@ object TracingTest extends DefaultRunnableSpec {
               )
             )
         },
+        testM("scopedEffect") {
+          for {
+            tracer <- ZIO.service[Tracer]
+            _ <- Tracing.scopedEffect {
+                  val span = tracer.getCurrentSpan
+                  span.addEvent("In legacy code")
+                  if (Context.current() == Context.ROOT) throw new RuntimeException("Current context is root!")
+                  Thread.sleep(10)
```

Comment:
Why do we need to sleep here? I'm not sure how `Context` works, but might it make sense to try to manipulate it from some other thread concurrently? From the test I can see no reason, why it should change.

Reply:
The test is probably equally valid without the sleep, it seemed like a reasonable way to ensure we didn't optimize out the one liner. We could try to gin up something stressing the concurrency but I'm not sure it's worth the effort in this context.

Reply:
OK. If you want to keep it, then please let's shorten it to a couple 100 ms. ;-)

Reply:
I've gone ahead and removed it, I think it's probably unnecessary and obviously an unpleasant code smell.